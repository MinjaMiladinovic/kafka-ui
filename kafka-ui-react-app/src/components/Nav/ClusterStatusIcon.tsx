import { ServerStatus } from 'generated-sources';
import React from 'react';
import { Colors } from 'theme/theme';

interface Props {
  status: ServerStatus;
}

const ClusterStatusIcon: React.FC<Props> = ({ status }) => {
  return (
    <svg
      width="4"
      height="4"
      viewBox="0 0 4 4"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="2"
        cy="2"
        r="2"
        fill={
          status === ServerStatus.ONLINE ? Colors.green[40] : Colors.red[50]
        }
      />
    </svg>
  );
};

export default ClusterStatusIcon;
