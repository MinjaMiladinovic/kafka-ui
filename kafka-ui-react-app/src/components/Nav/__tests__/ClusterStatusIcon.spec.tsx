import React from 'react';
import { mount } from 'enzyme';
import { ServerStatus } from 'generated-sources';
import ClusterStatusIcon from 'components/Nav/ClusterStatusIcon';

describe('ClusterStatusIcon', () => {
  describe('when online', () => {
    it('matches snapshot', () => {
      const wrapper = mount(<ClusterStatusIcon status={ServerStatus.ONLINE} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when offline', () => {
    it('matches snapshot', () => {
      const wrapper = mount(
        <ClusterStatusIcon status={ServerStatus.OFFLINE} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
