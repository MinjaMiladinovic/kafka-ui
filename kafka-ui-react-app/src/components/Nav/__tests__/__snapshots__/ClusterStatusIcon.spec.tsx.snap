// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterStatusIcon when offline matches snapshot 1`] = `
<ClusterStatusIcon
  status="offline"
>
  <svg
    fill="none"
    height="4"
    viewBox="0 0 4 4"
    width="4"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="2"
      cy="2"
      fill="#E51A1A"
      r="2"
    />
  </svg>
</ClusterStatusIcon>
`;

exports[`ClusterStatusIcon when online matches snapshot 1`] = `
<ClusterStatusIcon
  status="online"
>
  <svg
    fill="none"
    height="4"
    viewBox="0 0 4 4"
    width="4"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="2"
      cy="2"
      fill="#5CD685"
      r="2"
    />
  </svg>
</ClusterStatusIcon>
`;
